scipy/fft/tests/test_fftlog.py

```diff
@@ -149,8 +141,6 @@ def test_fht_special_cases(xp):
         assert record, 'ifht did not warn about a singular transform'
 
 
-# https://github.com/pytorch/pytorch/issues/59786
-@skip_if_array_api_backend('torch')
```

Comment:
I'm showing 28 extra test failures on this branch vs. `main` when using `cuda` backend via i.e., `SCIPY_DEVICE=cuda python dev.py test -j 32 -b all`, but I suspect this is just https://github.com/pytorch/pytorch/issues/106773. That is a bit annoying though, since there's no stable release with that patch yet, but if other devs don't mind too much I don't either, I think we've already done merges that ignore that previously.

As for running on the host, I can confirm the drop in number of skips:

- here:     `1 failed, 47792 passed, 4077 skipped, 149 xfailed, 12 xpassed in 125.33s (0:02:05)`
- `main`: `1 failed, 47752 passed, 4117 skipped, 149 xfailed, 12 xpassed in 133.21s (0:02:13)`

Reply:
This is indeed a little annoying. We don't have other failures in `cluster` or `fft` yet I believe. So given that this is not critical to merge and PyTorch 2.1.0 will be released on Oct 4th, how about we wait till then to merge this?

Reply:
> We don't have other failures in cluster or fft yet I believe.

This has however slipped in to `_lib._util._lazywhere`, with `TestLazywhere.test_basic` failing for this reason.

> PyTorch 2.1.0 will be released on Oct 4th, how about we wait till then to merge this?

Sounds sensible :+1: although maybe this doesn't matter since we're already failing in `_lib`. Note that my availability to contribute will be severely reduced during October and November. This PR looks like it should be okay without me though :crossed_fingers:.

This is causing failures in gh-19260, so I'll add some temporary GPU skips there if we are going to wait for the PyTorch release.

Reply:
Hmm okay, if it's going to cause a bunch of extra work and there are already failures, maybe it's fine to merge and we just have to just install a PyTorch nightly. I hadn't noticed `_lib`, just ran `test -s fft` and `test -s cluster`.

Reply:
PyTorch 2.1 is out now. Haven't tested the actual release yet, but this should be solved.